spec/javascripts/accordion-with-descriptions-spec.js

```diff
@@ -2,215 +2,242 @@ describe('An accordion with descriptions module', function () {
+        $subsectionButton.click();
+        expect($subsectionContent).not.toHaveClass("js-hidden");
+        $subsectionButton.click();
+        expect($subsectionContent).toHaveClass("js-hidden");
+      });
+
+      // When a section is closed (testing: toggleState, setExpandedState)
+      it("has a an aria-expanded attribute and the value is false", function () {
+        var $subsectionButton = $element.find('.subsection__title button:first');
+        var $subsectionContent = $element.find('.subsection__content');
+        $subsectionButton.click();
+        expect($subsectionButton).toHaveAttr('aria-expanded','true');
+        $subsectionButton.click();
+        expect($subsectionButton).toHaveAttr('aria-expanded','false');
+      });
+
+      it("has its state removed in session storage", function () {
+        var GOVUKServiceManualTopic = "GOVUK_service_manual_agile_delivery";
+
+        var $closedSubsections = $element.find('.subsection');
+        var subsectionClosedContentId = $closedSubsections.find('.subsection__content').attr('id');
+        sessionStorage.removeItem(GOVUKServiceManualTopic+subsectionClosedContentId , 'Opened');
+        var removedItem = sessionStorage.getItem(GOVUKServiceManualTopic+subsectionClosedContentId);
+        expect(removedItem).not.toExist();
+      });
 
-  describe('When a section is closed', function () {
-
-    // When a section is closed (testing: toggleSection, closeSection)
-    it("has a class of js-hidden", function () {
-      var $subsectionButton = $element.find('.subsection__title button:first');
-      var $subsectionContent = $element.find('.subsection__content:first');
-      $subsectionButton.click();
-      expect($subsectionContent).not.toHaveClass("js-hidden");
-      $subsectionButton.click();
-      expect($subsectionContent).toHaveClass("js-hidden");
     });
+  });
 
-    // When a section is closed (testing: toggleState, setExpandedState)
-    it("has a an aria-expanded attribute and the value is false", function () {
-      var $subsectionButton = $element.find('.subsection__title button:first');
-      var $subsectionContent = $element.find('.subsection__content');
-      $subsectionButton.click();
-      expect($subsectionButton).toHaveAttr('aria-expanded','true');
-      $subsectionButton.click();
-      expect($subsectionButton).toHaveAttr('aria-expanded','false');
+  describe('When linking to a topic section', function () {
+    beforeEach(function() {
+      var accordion = new GOVUK.Modules.AccordionWithDescriptions();
+      $element = $(html);
+      spyOn(GOVUK, 'getCurrentLocation').and.returnValue({
+        hash: '#topic-section-one'
+      });
+      accordion.start($element);
     });
 
-    it("has its state removed in session storage", function () {
-      var GOVUKServiceManualTopic = "GOVUK_service_manual_agile_delivery";
+    it("opens the linked to topic section", function () {
```

Comment:
You're right - that was it. It works.

When you click on a section to open it, the state is stored in the cookie so that the section is still open when you return to it. This isn't happening with these anchor links. It's probably not a super important feature and the user would be very unlikely to notice.. but it's slightly inconsistent. It might just be a case of calling the "remember the state" function.